core/app/helpers/spree/taxons_helper.rb

```diff
@@ -4,16 +4,26 @@ module TaxonsHelper
     # that we can use configurations as well as make it easier for end users to override this determination.  One idea is
     # to show the most popular products for a particular taxon (that is an exercise left to the developer.)
     def taxon_preview(taxon, max=4)
-      products = taxon.active_products.select("DISTINCT (spree_products.id), spree_products.*, spree_products_taxons.position").limit(max)
+      products = taxon.active_products.select(*selection_sql).limit(max)
       if (products.size < max)
         products_arel = Spree::Product.arel_table
         taxon.descendants.each do |taxon|
           to_get = max - products.length
-          products += taxon.active_products.select("DISTINCT (spree_products.id), spree_products.*, spree_products_taxons.position").where(products_arel[:id].not_in(products.map(&:id))).limit(to_get)
+          products += taxon.active_products.select(*selection_sql).where(products_arel[:id].not_in(products.map(&:id))).limit(to_get)
           break if products.size >= max
         end
       end
       products
     end
+
+    private
+
+    def selection_sql
+      [
+        "DISTINCT (spree_products.id)",
```

Comment:
This needs distinct because the scopes aren't set up right. It takes a currency in as an argument but doesn't do anything with it:

https://github.com/solidusio/solidus/blob/v1.0.0/core/app/models/spree/product/scopes.rb#L190-L193

It returns multiple records because records with multiple prices are returned by the scope if you have a product with a price in more than one currency.

If that's the way we want that to work, we should apply the distinct at that level. But really, I think we should properly respect currencies rather than just ignoring them.